estate_account/models/estate_property.py

```diff
@@ -0,0 +1,55 @@
+from odoo import models, api, Command
+
+
+class EstateProperty(models.Model):
+    _inherit = "estate.property"
+
+    def action_sold(self):
+        """ Override the action_sold method to create and post an invoice when a property is sold. """
+        print("✅ estate_account: action_sold overridden - Creating and Posting Invoice")  # Debugging
+
+        res = super().action_sold()
+
+        if not self.buyer_id:
+            raise ValueError("❌ Cannot create invoice: No buyer assigned to this property.")
+
+        journal = self.env['account.journal'].search([
+            ('type', '=', 'sale'),
+            ('company_id', '=', self.env.company.id)
+        ], limit=1)
+        if not journal:
+            raise ValueError(f"❌ No sales journal found for company {self.env.company.name}!")
+
+        default_income_account = self.env['account.account'].search([
+            ('account_type', 'in', ['income', 'income_other']),
+            ('company_ids', 'in', self.env.company.id)
+        ], limit=1)
+        if not default_income_account:
+            raise ValueError(f"❌ No valid income account found for company {self.env.company.name}!")
+
+        print(f"✅ Using Income Account: {default_income_account.id} - {default_income_account.name}")
+        selling_price = self.selling_price
```

Comment:
```suggestion
       // write the code to check access
        selling_price = self.selling_price
```
the function should check first whether the user has updation access or not